internal/services/elastic/elastic_monitor_resource.go

```diff
@@ -0,0 +1,444 @@
+package elastic
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/location"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/elastic/legacysdk/elastic/mgmt/2020-07-01/elastic"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/elastic/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/elastic/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceElasticMonitor() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceElasticMonitorCreate,
+		Read:   resourceElasticMonitorRead,
+		Update: resourceElasticMonitorUpdate,
+		Delete: resourceElasticMonitorDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.ElasticMonitorID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ElasticMonitorName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"elastic_properties": {
+				Type:     pluginsdk.TypeList,
```

Comment:
Please remove the properties that won't be referred in other azurerm resources.

Reply:
Even though some of these properties won't be referred in other azurerm resources, userr needs some of these for  data plane as well as some management operations (such kibana_sso_url). In this case we would want to export theses properties or is there any other preferred way ?

Reply:
Got it. Then how about moving it to root level?

Reply:
Updated